lib/jsonapi/serializer.ex

```diff
@@ -135,11 +135,19 @@ defmodule JSONAPI.Serializer do
     merge_related_links(data, info, remove_links?())
   end
 
-  defp merge_links(doc, data, view, conn, false) do
+  defp merge_links(doc, data, view, conn, false, true) do
+    links =
+      %{self: view.url_for(data, conn)}
+      |> Map.merge(view.links(data, conn))
```

Comment:
I was thinking of passing `doc.meta` to the `links` function as well, as it's probably the best way to send pagination data in. What do you think?

Reply:
I would err on the side of passing along less information than more until there is a specific need.

Reply:
How would you propose passing in pagination data such as a cursor, limit or offset then? I'm assuming that the controller (or whoever is serializing the data) will be in charge of that and it will be pretty opaque to the view. Am I seeing this wrong?

Reply:
In my case the pagination is part of the data so I have what I do.  Could you share your code where you need `doc`?

Reply:
I think it's also important to remain consistent in the code and we are passing `data, conn` for the other overridable functions.

Reply:
I'm not sure I need it. I'm still trying to figure out if I really do. Let's say I have a limit and offset I want to use, do I pass that as data in a map? Something like:

```elixir
render(conn, "index.json", %{
  data: %{
    collection: data,
    pagination: %{
      limit: limit,
      offset: offset
    }
  }
})
```

Reply:
@zamith that's how the Elixir pagination libraries I'm familiar with work.  You're also passing in the `conn` so you _could_ do something like `put_private(conn, :pagination, ...)`

On a side note, are you a time traveler?!  GitHub keeps sending me notifications about your notifications occurring _in the future_ 🤔 

<img width="261" alt="screen shot 2018-05-02 at 8 58 06 am" src="https://user-images.githubusercontent.com/73386/39531036-801cdc1c-4de7-11e8-8409-f98452789ff2.png">

Reply:
Well, that was my first approach, the problem is that the view assumes that `data` is an array of models and the serializer will then `map_reduce`s over that. From this line:

```elixir
def index(models, conn, _params, meta \\ nil), do: serialize(__MODULE__, models, conn, meta)
```

So, if anything other than an array is passed, the iteration is messed up. That's why I reached for the metadata as a way to pass in extra information. Does this make sense?

Well, I'm guessing timezones are to blame :)

Reply:
@doomspork There is still this unresolved issue, or do you think it's not an issue at all?